### DISCUSSION TITLE
How to Get a Parameter of a Field Input-Type Entity-Query

### DISCUSSION THREAD

**Discussion by MariaGanan** [2sic/2sxc#3254] [Q&A] [answered]
Hello,
We are doing an app and we have a question.
We have two fields, one "Category" and another "Subcategory", we want to select the "Category" first and then filter the related subcategories on the second one.
For the "Subcategory" field we are using "Field Input-Type entity-query" and we have the query but we don't know how to access the Parameter we put there.
![imagen](https://github.com/2sic/2sxc/assets/139103286/7f60837c-018e-4c79-a64c-57eb3743604f)

On the Query we have the Relationship Filter and we need to know how to acces the parameter.
![imagen](https://github.com/2sic/2sxc/assets/139103286/9a4c9a99-5567-43c4-b208-db541e9789e2)

Thanks,

**Comment by iJungleboy** [comment]
The parameter is passed as a url-parameter, so the query will be called with `category=[some-value]`

To access it in your query, use a token like `[QueryString:Category]` 

In this case, 
* `[...]` marks a placeholder
* `QueryString` means the source is the url-parameters
* `:` separator
* `Category` is the name of the url parameter

**Comment by MariaGanan** [comment]
HI,
Thanks for your answer, but I couldn't fix this.
"Category" Its an Entity, is this the problem? Should I pass only strings or something?
Thanks,

**Comment by iJungleboy** [comment, marked as answer]
I'm not sure what exactly you're doing, but I recommend:

1. First create the query and make sure it works with url-parameters
2. Then create the dropdown which uses a fixed value to run the query
3. Then optimize the dropdown to use the value from another field in the form

## Setup the Query

Basically you want to get a query which responds to the url parameter, and which you can test/verify all from the query designer. 
Steps are typically:

1. Create a basic query without parameters
2. Add "constant" parameters like in the screenshot above, set `Filter` to `Daniel` or whatever - until the query works. It's important to do this first, as any confusion as to what the value does and how it works will get bigger the larger the problem becomes. So basically the query should first work with constant values so you're sure it does what you expect.
5. Then replace the constant values with placeholders - eg `[QueryString:Category]` or something
6. Then **test it** from within the query designer, by adding test values (left side of the screen) - probably something like this: `[QueryString:Category]=Daniel`
7. If you run it now - with the test values, you should still get the same result as before with the constant value, and you're sure the query is good and ready to go

## Setup the Dropdown

1. again start with constant values - so set parameters to be something like `Category=Daniel` - test, and verify that you get what you expect
2. then update it to use `Category=[Fieldname-In-Form]` and again verify that everything works

Good luck 🍀!

**Comment by MariaGanan** [comment]
Hi,
All works untill I do the last step:
     "2. then update it to use Category=[Fieldname-In-Form] and again verify that everything works"

The Category field it's an entity / item picker, my question is, it can be done with an entity? 

Thanks

**Comment by iJungleboy** [comment]
could you please

1. post a print-screen of your entire query
2. briefly describe what exactly your trying to do/filter so I can better answer this ;)

**Comment by MariaGanan** [comment]
I have an app called “Library” with 3 content types:

CATEGORY
- Title (String basic)
- Picture (Link to Url, File or Page)
- Description (String wysiwyg)
- Urlkey (Url-Path)

SUBCATEGORY
- Title (String basic)
- Category (Entity/Item picker)
- Urlkey (Url-Path)

ITEM
- Title (String basic)
- Category (Entity/Item picker)
- Subcategory(Entity/Query based entity/ Item-picker)
- Description (String wysiwyg)
- Urlkey  (Url-Path)

I want that when you go to create an Item if you select a Category then only Subcategories with this Category will show on the dropdown for Subcategory.

Thank you so much!

![Subcategory](https://github.com/2sic/2sxc/assets/139103286/de356580-5159-4c9b-a9d3-a9756de65f0c)
![Item](https://github.com/2sic/2sxc/assets/139103286/f7dcd518-c4ac-4e0c-97d2-b64edc7c405d)
![Query](https://github.com/2sic/2sxc/assets/139103286/75fb2704-8637-4e99-ae3d-3a404e0eb65c)

**Comment by MariaGanan** [comment]
Hi @iJungleboy  Thanks for all your Help!!! I solved IT!
I was using a wrong field type, now with the dropdown with the query it works.
Again, Thank you very much for your patience!

**Reply by iJungleboy** [reply]
you're welcome ;)
